app/api/probe_dock/results_api.rb

```diff
@@ -34,6 +34,17 @@ def current_organization
             Organization.active.joins(projects: :tests).where('project_tests.api_id = ?', params[:testId]).first!
           end
         end
+
+        def serialization_options(results)
+          # Make sure this will not process the retrieval of payloads multiple times
+          return @serialization_options if @serialization_options
+
+          # Serialization options with the payloads containing the SCM data
+          @serialization_options = {
+            with_scm: true_flag?(:withScm),
+            payloads: TestPayload.with_scm_data.where('test_payloads.id in (?)', results.collect(&:test_payload_id))
```

Comment:
I think it would be good to already handle the case where `results` is a single result, as you did in `tests_api`.